python/ray/ml/predictors/integrations/lightgbm/lightgbm_predictor.py

```diff
@@ -99,4 +106,13 @@ def predict(
             DataBatchType: Prediction result.
 
         """
-        raise NotImplementedError
+        predict_kwargs = predict_kwargs or {}
+
+        data = self.preprocessor.transform_batch(data)
```

Comment:
Agree with @matthewdeng's point- this won't work if self.preprocessor is None, and I don't think we should require a preprocessor.

Reply:
And can we also add a test for this?